StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TriviaHelper.cs

```diff
@@ -95,5 +97,71 @@ internal static int IndexOfTrailingWhitespace(SyntaxTriviaList triviaList)
 
             return (whiteSpaceStartIndex < triviaList.Count) ? whiteSpaceStartIndex : -1;
         }
+
+        /// <summary>
+        /// Strips all trialing whitespace trivia from the trivia list until a non-whitespace trivia is encountered.
+        /// end-of-line trivia are ignored, but kept.
+        /// </summary>
+        /// <param name="triviaList">The trivia list to strip of its trailing whitespace.</param>
+        /// <returns>The modified triviaList.</returns>
+        internal static SyntaxTriviaList WithoutTrailingWhitespace(this SyntaxTriviaList triviaList)
+        {
+            var result = new List<SyntaxTrivia>();
+            bool skipWhitespace = true;
+            for (var index = triviaList.Count - 1; index >= 0; index--)
+            {
+                var trivia = triviaList[index];
+                switch (trivia.Kind())
+                {
+                    case SyntaxKind.WhitespaceTrivia:
+                        if (!skipWhitespace)
+                        {
+                            result.Insert(0, trivia);
+                        }
+                        break;
+                    case SyntaxKind.EndOfLineTrivia:
+                        result.Insert(0, trivia);
+                        break;
+                    default:
+                        result.Insert(0, trivia);
+                        skipWhitespace = false;
+                        break;
+                }
+            }
+            return SyntaxFactory.TriviaList(result);
+        }
+
+        /// <summary>
+        /// Strips all leading whitespace trivia from the trivia list until a non-whitespace trivia is encountered.
+        /// end-of-line trivia are ignored, but kept.
+        /// </summary>
+        /// <param name="triviaList">The triviaList to strip of its leading whitespace.</param>
+        /// <returns>The modified triviaList.</returns>
+        internal static SyntaxTriviaList WithoutLeadingWhitespace(this SyntaxTriviaList triviaList)
```

Comment:
I dont like the method name either. See my comment on the other method.

Using List should be okay here.